File: plugin-template/ios/Plugin/Plugin/MyPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitor.ionicframework.com/docs/ios/plugins
 *
 * Important: the name of your plugin class *must* be globally unique to the Capacitor
 * ecosystem!
 */
@objc(MyPlugin)
public class MyPlugin : CAPPlugin {
    
    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.success([
            "value": value
        ])
    }
}

File: plugin-template/android/plugin/src/main/java/com/getcapacitor/myplugin/MyPlugin.java
package com.getcapacitor.myplugin;

import com.getcapacitor.JSObject;
import com.getcapacitor.NativePlugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;

@NativePlugin()
public class MyPlugin {

    @PluginMethod()
    public void echo(PluginCall call) {
        String value = call.getString("value");

        JSObject ret = new JSObject();
        ret.put("value", value);
        call.success(ret);
    }
}
